framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -723,12 +732,46 @@ protected void addInformationFromPreconditions(S info,
 
         S info = in.getRegularStore();
         V rhsValue = in.getValueOfSubNode(rhs);
+        Receiver expr = FlowExpressions.internalReprOf(
+                analysis.getTypeFactory(), lhs);
+        if (performWholePrograminference(
+                n.getTree(), InternalUtils.symbol(lhs.getTree()))) {
+            if (expr instanceof FieldAccess) {
```

Comment:
You are right. I'm now using `lhs instanceof FieldAccessNode` and removed `expr` in both locations. I pushed the changes.